File: phi_plugin/model/getBanGroup.py
from ..models import phigrosBanGroup
from .cls.type import allFnc


class getBanGroup:
    @staticmethod
    async def get(group: str | None, fnc: allFnc) -> bool:
        if not group:
            return False
        match fnc:
            case "help" | "tkhelp":
                return await phigrosBanGroup.getStatus(group, "help")
            case "bind" | "unbind":
                return await phigrosBanGroup.getStatus(group, "bind")
            case (
                "b19"
                | "arcgrosB19"
                | "update"
                | "info"
                | "list"
                | "singlescore"
                | "lvscore"
                | "chap"
                | "suggest"
            ):
                return await phigrosBanGroup.getStatus(group, "b19")
            case "bestn" | "data":
                return await phigrosBanGroup.getStatus(group, "wb19")
            case "song" | "ill" | "search" | "alias" | "randmic":
                return await phigrosBanGroup.getStatus(group, "song")
            case "rankList" | "godList":
                return await phigrosBanGroup.getStatus(group, "ranklist")
            case "comrks" | "tips" | "lmtAcc" | "randClg":
                return await phigrosBanGroup.getStatus(group, "fnc")
            case "tipgame":
                return await phigrosBanGroup.getStatus(group, "tipgame")
            case "guessgame":
                return await phigrosBanGroup.getStatus(group, "guessgame")
            case "ltrgame":
                return await phigrosBanGroup.getStatus(group, "ltrgame")
            case "sign" | "send" | "tasks" | "retask" | "jrrp":
                return await phigrosBanGroup.getStatus(group, "sign")
            case "theme":
                return await phigrosBanGroup.getStatus(group, "setting")
            case "dan" | "danupdate":
                return await phigrosBanGroup.getStatus(group, "dan")
            case _:
                return False

File: phi_plugin_koishi/model/getBanGroup.ts
import { redis, banGroupId } from '../components/redis'
import { allFnc } from "./type/type"
import { Keys } from "koishi"

export default class getBanGroup {

    /**
     * 
     * @param {string} group 
     * @param {string} fnc 
     * @returns 
     */
    static async redis(group: string, fnc: string) {
        let res = await redis.get('phigrosBanGroup', { groupId: group })
        return res[0] && res[0][fnc] ? true : false
    }

    /**
     * 
     * @param {string} group 
     * @param {string} fnc 
     * @returns 
     */
    static async get(group: string, fnc: allFnc) {
        if (!group) {
            return false
        }
        switch (fnc) {
            case 'help':
            case 'tkhelp':
                return await this.redis(group, 'help')
            case 'bind':
            case 'unbind':
                return await this.redis(group, 'bind')
            case 'b19':
            case 'arcgrosB19':
            case 'update':
            case 'info':
            case 'list':
            case 'singlescore':
            case 'lvscore':
            case 'chap':
            case 'suggest':
                return await this.redis(group, 'b19')
            case 'bestn':
            case 'data':
                return await this.redis(group, 'wb19')
            case 'song':
            case 'ill':
            case 'search':
            case 'alias':
            case 'randmic':
                return await this.redis(group, 'song')
            case 'rankList':
            case 'godList':
                return await this.redis(group, 'ranklist')
            case 'comrks':
            case 'tips':
            case 'lmtAcc':
            case 'randClg':
                return await this.redis(group, 'fnc')
            case 'tipgame':
                return await this.redis(group, 'tipgame')
            case 'guessgame':
                return await this.redis(group, 'guessgame')
            case 'ltrgame':
                return await this.redis(group, 'ltrgame')
            case 'sign':
            case 'send':
            case 'tasks':
            case 'retask':
            case 'jrrp':
                return await this.redis(group, 'sign')
            case 'theme':
                return await this.redis(group, 'setting')
            case 'dan':
            case 'danupdate':
                return await this.redis(group, 'dan')
            default:
                return false;
        }
    }
}